File: android/app/src/main/java/app/footycity/app/plugins/CapacitorPluginExample.kt
package app.capacitorstarter.app.plugins

import com.getcapacitor.JSObject
import com.getcapacitor.Plugin
import com.getcapacitor.PluginCall
import com.getcapacitor.PluginMethod
import com.getcapacitor.annotation.CapacitorPlugin

@CapacitorPlugin(name = "CapacitorPluginExample")
class CapacitorPluginExample : Plugin() {
    @PluginMethod
    fun echo(call: PluginCall) {
        val value = call.getString("value")
        val ret = JSObject()
        ret.put("value", value)
        ret.put("source", "hello from android!")
        println("hello from android!")
        call.resolve(ret)
    }
}

File: ios/App/App/CapacitorPluginExample.swift
import Capacitor

@objc(CapacitorPluginExample)
public class CapacitorPluginExample: CAPPlugin {
    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve(["value": value])
    }
}
